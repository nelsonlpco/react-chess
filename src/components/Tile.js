import React,{Component} from 'react';
import PropTypes from 'prop-types';

class Tile extends Component
{ 
    setTileClass = (tileLine, tileColumn) =>(
        (tileLine + tileColumn) % 2 === 0  ? 'black-tile' : 'white-tile'
    );
    
    shouldComponentUpdate = (nextProps, nextState) => {
        return nextProps.children !== this.props.children;
    }
    
    render(){
        var {children, y, x} = this.props;

        return (
            <div className={`tile ${this.setTileClass(y, x)}` }>
                {children}
            </div>
        );
    }
};


Tile.propTypes = {
    x : PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
}

export default Tile;