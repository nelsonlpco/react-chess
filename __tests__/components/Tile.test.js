import React from 'react';
import {shallow} from 'enzyme';
import Tile from './../../src/components/Tile';

describe('Tile component tests', () => {
    it('rendering tile component without width and height', () => {
        var wrapper = shallow(<Tile color='#fff000' />);

        expect(wrapper.prop('style').width).toEqual(undefined);
        expect(wrapper.prop('style').height).toEqual(undefined);
        expect(wrapper).toMatchSnapshot();
    });

    it('rendering tile component with width and height', () => {
        var wrapper = shallow(<Tile width={400} height={400} color='#fff000' />);

        expect(wrapper.prop('style').width).toEqual('400px');
        expect(wrapper.prop('style').height).toEqual('400px');
        expect(wrapper).toMatchSnapshot();
    });
})
