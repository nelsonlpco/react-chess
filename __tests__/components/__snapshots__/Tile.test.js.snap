// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component tests rendering tile component with width and height 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tile
    color="#fff000"
    height={400}
    width={400}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "style": Object {
        "color": "#fff000",
        "height": "400px",
        "width": "400px",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "style": Object {
          "color": "#fff000",
          "height": "400px",
          "width": "400px",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Tile component tests rendering tile component without width and height 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tile
    color="#fff000"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "style": Object {
        "color": "#fff000",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "style": Object {
          "color": "#fff000",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
